tests/test_sinope.py

```diff
@@ -57,3 +73,135 @@ async def test_sinope_flow_measurement(zigpy_device_from_quirk, quirk):
+    zigpy_device_from_quirk, quirk, press_type, exp_event
+):
+    """Test that button presses are sent as events."""
+    device: Device = zigpy_device_from_quirk(quirk)
+    cluster_id = SINOPE_MANUFACTURER_CLUSTER_ID
+    endpoint_id = 1
+
+    class Listener:
+        zha_send_event = mock.MagicMock()
+
+    cluster_listener = Listener()
+    device.endpoints[endpoint_id].in_clusters[cluster_id].add_listener(cluster_listener)
+
+    attr = foundation.Attribute(
+        attrid=0x54,  # "action_report" attribute
+        value=foundation.TypeValue(
+            type=t.enum8(0x30),
+            value=press_type,
+        ),
+    )
+    data = _get_packet_data(foundation.GeneralCommand.Report_Attributes, attr)
+    device.handle_message(260, cluster_id, endpoint_id, endpoint_id, data)
+
+    if exp_event is None:
+        assert cluster_listener.zha_send_event.call_count == 0
+    else:
+        assert cluster_listener.zha_send_event.call_count == 1
+        assert cluster_listener.zha_send_event.call_args == mock.call(
+            exp_event,
+            {
+                "attribute_id": 84,
+                "attribute_name": "action_report",
+                "value": press_type.value,
+            },
+        )
+
+
+@pytest.mark.parametrize("quirk", (SinopeTechnologieslight,))
+async def test_sinope_light_switch_non_action_report(zigpy_device_from_quirk, quirk):
+    """Test commands not handled by custom handler.
+
+    Make sure that non attribute report commands and attribute reports that don't
+    concern action_report are passed through to base class.
+    """
+
+    device: Device = zigpy_device_from_quirk(quirk)
+    cluster_id = SINOPE_MANUFACTURER_CLUSTER_ID
+    endpoint_id = 1
+
+    class Listener:
+        zha_send_event = mock.MagicMock()
+
+    cluster_listener = Listener()
+    device.endpoints[endpoint_id].in_clusters[cluster_id].add_listener(cluster_listener)
+
+    # read attributes general command
+    data = _get_packet_data(foundation.GeneralCommand.Read_Attributes)
+    device.handle_message(260, cluster_id, endpoint_id, endpoint_id, data)
+    # no ZHA events emitted because we only handle Report_Attributes
+    assert cluster_listener.zha_send_event.call_count == 0
+
+    # report attributes command, but not "action_report"
+    attr = foundation.Attribute(
+        attrid=0x10,  # "on_intensity" attribute
+        value=foundation.TypeValue(
+            type=t.int16s(0x29), value=t.int16s(50)
+        ),  # 0x29 = t.int16s
+    )
+    data = _get_packet_data(foundation.GeneralCommand.Report_Attributes, attr)
+    device.handle_message(260, cluster_id, endpoint_id, endpoint_id, data)
+    # ZHA event emitted because we pass non "action_report"
+    # reports to the base class handler.
+    assert cluster_listener.zha_send_event.call_count == 1
+
+
+@pytest.mark.parametrize("quirk", (SinopeTechnologieslight,))
+async def test_sinope_light_switch_reporting(zigpy_device_from_quirk, quirk):
```

Comment:
Tbh, I'm not sure we really need this test in zha quirks? We're not really testing any quirk specific code here, right?
(maybe except that the attribute exists)

We can leave this in for now though.